kotlin/workflow-ui-android/src/main/java/com/squareup/workflow/ui/ViewBackHandler.kt

```diff
@@ -0,0 +1,65 @@
+package com.squareup.workflow.ui
+
+import android.content.Context
+import android.content.ContextWrapper
+import android.view.View
+import android.view.View.OnAttachStateChangeListener
+import androidx.activity.ComponentActivity
+import androidx.activity.OnBackPressedCallback
+
+typealias BackPressHandler = () -> Unit
+
+var View.backPressedHandler: BackPressHandler?
+  get() {
+    val handler = getTag(R.id.view_back_handler)
+    check(handler is Function0<*>?) {
+      "Expected backPressedHandler of $this to be a Function0, found $handler."
+    }
+
+    @Suppress("UNCHECKED_CAST")
+    return handler as? BackPressHandler
+  }
+  set(value) {
+    setTag(R.id.view_back_handler, value)
+    if (value == null) {
+      onBackPressedCallback?.let { removeOnAttachStateChangeListener(it) }
+      onBackPressedCallback = null
+    } else {
+      if (onBackPressedCallback == null) onBackPressedCallback =
+        AttachedOnBackPressedCallback(this).apply {
+          addOnAttachStateChangeListener(this)
+        }
+    }
+  }
+
+private var View.onBackPressedCallback: AttachedOnBackPressedCallback?
+  get() {
+    return getTag(R.id.view_on_back_pressed_callback) as? AttachedOnBackPressedCallback
+  }
+  set(value) {
+    onBackPressedCallback?.remove()
```

Comment:
Needs a comment explaining why this is necessary - I think I know, but so many things are being added and removed in this file that I feel like I'm reading a riddle.